junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/TestMethodTestDescriptor.java

```diff
@@ -230,4 +234,48 @@ private void invokeAfterEachCallbacks(JupiterEngineExecutionContext context) {
 		});
 	}
 
+	@Override
+	public void nodeSkipped(JupiterEngineExecutionContext context, TestDescriptor descriptor, SkipResult result) {
+		invokeTestWatchers(context, t -> {
```

Comment:
I think test watchers should be invoked in "registration order" for skipped events since a node is skipped _before_ it would have been executed (analogous to before-each lifecycle callbacks); however, your current implementation of `invokeTestWatchers()` retrieves the watchers via `registry.getReversedExtensions()` which will always invoke the watchers in "reverse registration order".

@junit-team/junit-lambda, thoughts?